Switch `bosh instances` to the /instances endpoint

Kiemes opened the pull request:
The instances command of the go cli does now call /instances instead of /vms.
If the director does not support /instances (< v256) we fall back to /vms.

Additionally we renamed `State` to `Process State` and added a `State` column to the instances output which reflect the instance state. All according to @cppforlife feedback in https://www.pivotaltracker.com/story/show/114101171

cppforlife commented on director/instances.go:
```diff
@@ -0,0 +1,25 @@
+func (d DeploymentImpl) InstanceInfos() ([]VMInfo, error) {
+	infos, err := d.client.DeploymentInstanceInfos(d.name)
+
+	if err != nil && strings.Contains(err.Error(), "404") {
```
a bit sketchy to check for "404". i actually think we can just require director to be new enough to have /instances, so i dont think we need to fallback. prolly in future have to introduce first class type of error to handle fallback cases.

cppforlife commented on director/instances_test.go:
```diff
@@ -0,0 +1,226 @@
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"strings"
```
system libraries go at the top, separated by newline.

cppforlife commented on director/vms.go:
```diff
@@ -109,26 +110,35 @@ func (d DeploymentImpl) VMInfos() ([]VMInfo, error) {
+	return c.DeploymentResourceInfos(deploymentName, "vms")
+}
+
+func (c Client) DeploymentResourceInfos(deploymentName string, resourceType string) ([]VMInfo, error) {
```
seems like DeploymentResourceInfos should be private given that only DeploymentVMInfos and DeploymentInstanceInfo should call it.

cppforlife commented on director/vms.go:
```diff
@@ -109,26 +110,35 @@ func (d DeploymentImpl) VMInfos() ([]VMInfo, error) {
+	path := fmt.Sprintf("/deployments/%s/%s?format=full", deploymentName, resourceType)
 
 	_, resultBytes, err := c.taskClientRequest.GetResult(path)
+
```
remove newline

cppforlife commented on director/vms_test.go:
```diff
@@ -32,6 +32,7 @@ var _ = Describe("VMs", func() {
 	})
 
 	Describe("VMInfos", func() {
+
```
remove newline.

cppforlife commented on director/vms.go:
```diff
@@ -109,26 +110,35 @@ func (d DeploymentImpl) VMInfos() ([]VMInfo, error) {
 		return nil, err
 	}
 
+	addTimestampToInfos(infos)
```
prolly better to move it into DeploymentResourceInfos and just do it once.

Kiemes commented:
@cppforlife Thx for your feedback. We incorporated it. While doing this, we saw that VMInfo.Timestamp was never used and also not set correctly, so we removed it.
